pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -86,6 +87,8 @@ type Range struct {
 	family api.IPFamily
 
 	alloc allocator.Interface
+	// metrics is a metrics recorder that can be disabled
+	metrics metricsRecorderInterface
```

Comment:
I like this solution and seems aligned with the rest of the code base, 
@thockin what do you think?

Reply:
@dgrisonnet (sig-instrumentation) I'd like your opinion too, the fact that metrics are global is :grimacing: ,  what do you think about this solution ?

Reply:
I think that current proposal is good enough, maybe is a TODO for sig-instrumentation to work on standardize this ;)

Reply:
Yeah, it is good enough, we haven't standardized anything for now, so the implementation is up to the component owners.

That's a good point, I'll bring it to the group.